### PR TITLE
Platform secrets

### PR DESCRIPTION
**Pull request by facchettos**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind feature

**What does this pull request do? Which issues does it resolve?** (use `resolves #<issue_number>` if possible) 
resolves eng-3731
depends on #1779 

**Please provide a short message that should be published in the vcluster release notes**
Fixed an issue where vcluster ...

**What else do we need to know?**

### CODE REVIEW

File: cmd/vclusterctl/cmd/platform/get/get.go [hunk 1]
```diff
@@ -19,5 +19,6 @@ func NewGetCmd(globalFlags *flags.GlobalFlags, cfg *config.CLI) *cobra.Command {
 	}
 
 	cmd.AddCommand(newClusterCmd(globalFlags, cfg))
+	cmd.AddCommand(NewSecretCmd(globalFlags, defaults))
```
Comment (ThomasK33) [inline cmd/vclusterctl/cmd/platform/get/get.go:22]: If `NewSecretCmd` is public, shouldn't `newClusterCmd` also be public?

File: cmd/vclusterctl/cmd/platform/set/secret.go [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+	if namespace == "" {
+		namespace = "loft"
+	}
+
+	return namespace, nil
+}
+
+func GetProjectSecretNamespace(project string) (string, error) {
+	return ProjectNamespacePrefix + project, nil
+}
```
Comment (ThomasK33) [inline cmd/vclusterctl/cmd/platform/set/secret.go:281]: This function doesn't work with the latest platform version.
Project namespace prefixes must be fetched at runtime and included in the selves response.

**Review by ThomasK33 — CHANGES_REQUESTED** [review]

File: cmd/vclusterctl/cmd/platform/get/get.go (hunk 1 above)
Comment (facchettos) [reply cmd/vclusterctl/cmd/platform/get/get.go:22]: no, it should be private too, just copied over from the loftctl code

File: cmd/vclusterctl/cmd/platform/set/secret.go (hunk 2 above)
Comment (facchettos) [reply cmd/vclusterctl/cmd/platform/set/secret.go]: yeah good catch I missed a commit from loftctl

File: cmd/vclusterctl/cmd/platform/set/secret.go [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+)
+
+var (
+	ErrNoSecret = errors.New("please specify a secret key to set")
+)
+
+type SecretType string
+
+const (
+	ProjectNamespacePrefix            = "loft-p-"
```
Comment (ThomasK33) [inline cmd/vclusterctl/cmd/platform/set/secret.go:33]: This constant is unused

File: cmd/vclusterctl/cmd/root.go [hunk 4]
```diff
@@ -87,6 +90,10 @@ func BuildRoot(log log.Logger) (*cobra.Command, error) {
 	persistentFlags := rootCmd.PersistentFlags()
 	globalFlags = flags.SetGlobalFlags(persistentFlags, log)
 	cfg := globalFlags.LoadedConfig(log)
+	defaults, err := pdefaults.NewFromPath(pdefaults.ConfigFolder, pdefaults.ConfigFile)
+	if err != nil {
+		log.Debugf("Error loading defaults: %v", err)
```
Comment (ThomasK33) [inline cmd/vclusterctl/cmd/root.go:95]: Shouldn't we exit here if default loading fails?

**Review by ThomasK33 — CHANGES_REQUESTED** [review]

**Review by ThomasK33 — APPROVED** [review]

**Pull request merged**